app/lib/foreman_deployments/tasks/host_creation_task_definition.rb

```diff
@@ -0,0 +1,39 @@
+module ForemanDeployments
+  module Tasks
+    class HostCreationTaskDefinition < ForemanDeployments::Tasks::CreationTaskDefinition
+      class Action < BaseAction
+        def run
+          obj = HostCreationTaskDefinition.create_object(input)
+          obj.save!
+          HostCreationTaskDefinition.create_output(obj, output)
+        end
+      end
+
+      def self.create_object(parameters)
+        object_type = parameters['class']
+        object_params = parameters['params']
+        object_params['managed'] ||= true
+
+        object_type = object_type.constantize if object_type.is_a? String
+
+        object = set_parameters(object_type.new, object_params)
+
+        if object.is_a? Host::Managed
+          object_params['build'] ||= true
+          # Set attributes from a hostgroup
+          object.attributes = object.apply_inherited_attributes(object_params)
```

Comment:
You don't have to call `apply_inherited_attributes` explicitly, if the host is a new object. It is called implicitly [here](https://github.com/theforeman/foreman/blob/develop/app/models/host/managed.rb#L48).
I can thank @domcleal for this change - he insisted on a seamless integration of this feature.